dataprovider/dataprovider.go

```diff
@@ -36,6 +36,11 @@ import (
 
 const subscribeChannelSize = 32
 
+const (
+	adapatersCount  = 8
```

Comment:
numPreallocatedAdapters
numPreallocatedPathes 

or something like that.

Reply:
done